salt/modules/xbps-pkg.py

```diff
@@ -0,0 +1,641 @@
+
+
+def remove(name=None, pkgs=None, recursive=True, **kwargs):
+    '''
+    name
+        The name of the package to be deleted.
+
+    recursive
+        Also remove dependant packages (not required elsewhere).
+        Default mode: enabled.
+
+    Multiple Package Options:
+
+    pkgs
+        A list of packages to delete. Must be passed as a python list. The
+        ``name`` parameter will be ignored if this option is passed.
+
+    Returns a list containing the removed packages.
+
+    .. versionadded:: XXX 201X.XX
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' pkg.remove <package name> [recursive=False]
+        salt '*' pkg.remove <package1>,<package2>,<package3> [recursive=False]
+        salt '*' pkg.remove pkgs='["foo", "bar"]' [recursive=False]
+    '''
+
+    try:
+        pkg_params, pkg_type = __salt__['pkg_resource.parse_targets'](
+            name, pkgs
+        )
+    except MinionError as exc:
+        raise CommandExecutionError(exc)
+
+    if not pkg_params:
+        return {}
+
+    old = list_pkgs()
+    
+    # keep only installed packages
+    targets = [x for x in pkg_params if x in old]
+    if not targets:
+        return {}
+
+    cmd = ['xbps-remove', '-y']
+    if recursive:
+        cmd.append('-R')
+    cmd.extend(targets)
+    __salt__['cmd.run'](cmd, output_loglevel='trace')
+    __context__.pop('pkg.list_pkgs', None)
+    new = list_pkgs()
+
+    return salt.utils.compare_dicts(old, new)
+
+
+def list_repos():
+    '''
+    List all repos known by XBPS
+
+    .. versionadded:: XXX 201X.XX
+
+    CLI Example:
+
+    .. code-block:: bash
+
+       salt '*' pkg.list_repos
+    '''
+    repos = {}
+    out = __salt__['cmd.run']('xbps-query -L', output_loglevel='trace')
+    for line in out.splitlines():
+        repo = {}
+        if not line:
+            continue
+        try:
+            nb, url, rsa = line.strip().split(' ',2)
+        except ValueError:
+            log.error('Problem parsing xbps-query: Unexpected formatting in '
+                      'line: "{0}"'.format(line))
+        repo['nbpkg'] = int(nb) if nb.isdigit() else 0
+        repo['url'] = url
+        repo['rsasigned'] = True if rsa == '(RSA signed)' else False
+        repos[repo['url']] = repo
+    return repos
+
+
+def get_repo(repo, **kwargs):
+    '''
+    Display information about the repo.
+
+    .. versionadded:: XXX 201X.XX
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt '*' pkg.get_repo 'repo-url'
+    '''
+    repos = list_repos()
+    if repo in repos:
+        return repos[repo]
+    return {}
+
+
+def _locate_repo_files(repo, rewrite=False):
+    '''
+    Find what file a repo is called in.
+
+    Helper function for add_repo() and del_repo()
+
+    repo
+        url of the repo to locate (persistent).
+
+    rewrite
+        Whether to remove matching repository settings during this process.
+
+    Returns a list of absolute paths.
+    '''
+
+    import glob, re
```

Comment:
Why are we doing late imports here? These can safely be imported at the top of the file.